src/proto/h1/io.rs

```diff
@@ -325,35 +321,33 @@ impl ReadStrategy {
     }
 
     fn record(&mut self, bytes_read: usize) {
-        match *self {
-            ReadStrategy::Adaptive {
-                ref mut decrease_now,
-                ref mut next,
-                max,
-                ..
-            } => {
-                if bytes_read >= *next {
-                    *next = cmp::min(incr_power_of_two(*next), max);
-                    *decrease_now = false;
-                } else {
-                    let decr_to = prev_power_of_two(*next);
-                    if bytes_read < decr_to {
-                        if *decrease_now {
-                            *next = cmp::max(decr_to, INIT_BUFFER_SIZE);
-                            *decrease_now = false;
-                        } else {
-                            // Decreasing is a two "record" process.
-                            *decrease_now = true;
-                        }
-                    } else {
-                        // A read within the current range should cancel
-                        // a potential decrease, since we just saw proof
-                        // that we still need this size.
+        if let ReadStrategy::Adaptive {
+            ref mut decrease_now,
+            ref mut next,
+            max,
+            ..
+        } = *self
+        {
```

Comment:
Does rustfmt suggest this? I personally prefer braces after the expression instead of a newline usually...

Reply:
yeah this is rustfmt. I notice that this project doesn't have a `.rustfmt.toml`, this could be configured to avoid the newline in cases like this.